plugins/gtav/structs.h

```diff
@@ -0,0 +1,227 @@
+// Copyright 2022 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef GTAV_STRUCTS
+#define GTAV_STRUCTS
+
+#include <cstdint>
+
+// Reference: https://github.com/Yimura/GTAV-Classes
+
+typedef uint64_t ptr_t;
+
+enum class GameState : int32_t {
+	Invalid = -1,
+	Playing,
+	Died,
+	Arrested,
+	FailedMission,
+	LeftGame,
+	Respawn,
+	InMPCutscene
+};
+
+enum class ModelType : uint8_t { Unk0, Object, Unk1, Unk2, Weapon, Vehicle, Ped, Plant = 129 };
+
+namespace rage {
+template< typename T > union vector3 {
+	T data[3];
+	struct {
+		T x, y, z;
+	};
+};
```

Comment:
Afaik this is undefined behavior.

See also https://stackoverflow.com/questions/702658/c-union-array-and-vars

Besides: We should prefer `std::array` for fixed-sized arrays over plain C arrays.

Reply:
@Yimura

Reply:
I should add that the reason this might end up being undefined behavior (as usually is the case with unions) is that padding and data alignment can mess up the idea of corresponding the array entries with the struct entries.

Who is Yimura though?

Reply:
This union is just me experimenting a bit, you can just remove this union from the PR and replace the rage::fvector3 with a struct.

```cpp
struct fvector3 {
    float x, y, z;
}
```

Or any sort of typing that is 12 bytes large.
```cpp
typedef float[3] fvector3;
typedef std::array<float, 3> fvector3;
```

Whichever your guidelines fits better.

Reply:
I'll probably replace it with `std::array` and use `std::copy` in `mumble_fetchPositionalData()` instead of assignments.

Reply:
Just a reminder that this still requires a resolution